core/servers/chat/mrc_multiplexer.js

```diff
@@ -0,0 +1,269 @@
+};
+
+const connectedSockets = new Set();
+let mrcCentralConnection = '';
+
+exports.getModule = class MrcModule extends ServerModule {
+    constructor() {
+        super();
+        this.log        = Log.child( { server : 'MRC' } );
+    }
+
+    createServer(cb) {
+        if (!this.enabled) {
+            return cb(null);
+        }
+
+        const self = this;
+
+        const config = Config();
+        const boardName  = config.general.prettyBoardName || config.general.boardName;
+        const enigmaVersion = 'ENiGMA-BBS_' + require('../../../package.json').version;
+
+        const mrcConnectOpts = {
+            host    : config.chatServers.mrc.serverHostname || 'mrc.bottomlessabyss.net',
+            port    : config.chatServers.mrc.serverPort || 5000
+        };
+
+        const handshake = `${boardName}~${enigmaVersion}/${os.platform()}-${os.arch()}/${PROTOCOL_VERSION}`;
+        this.log.debug({ handshake : handshake }, 'Handshaking with MRC server');
+
+        // create connection to MRC server
+        this.mrcClient = net.createConnection(mrcConnectOpts, () => {
+            this.mrcClient.write(handshake);
+            this.log.info(mrcConnectOpts, 'Connected to MRC server');
+            mrcCentralConnection = this.mrcClient;
+        });
+
+        // do things when we get data from MRC central
+        this.mrcClient.on('data', (data) => {
+            // split on \n to deal with getting messages in batches
+            data.toString().split('\n').forEach( item => {
+                if (item == '') return;
+
+                this.log.debug( { data : item } , 'Received data');
+                let message = this.parseMessage(item);
+                this.log.debug(message, 'Parsed data');
+                if (message) {
+                    this.receiveFromMRC(this.mrcClient, message);
+                }
+            });
+        });
+
+        this.mrcClient.on('end', () => {
+            this.log.info(mrcConnectOpts, 'Disconnected from MRC server');
+        });
+
+        this.mrcClient.on('error', err => {
+            this.log.info( { error : err.message }, 'MRC server error');
+        });
+
+        // start a local server for clients to connect to
+        this.server = net.createServer( function(socket) {
+            socket.setEncoding('ascii');
+
+            socket.on('data', data => {
+                // split on \n to deal with getting messages in batches
+                data.toString().split('\n').forEach( item => {
+                    if (item == '') return;
+
+                    // save username with socket
+                    if(item.startsWith('--DUDE-ITS--')) {
+                        connectedSockets.add(socket);
+                        socket.username = item.split('|')[1];
+                        Log.debug( { server : 'MRC', user: socket.username } , 'User connected');
+                    } else {
+                        self.receiveFromClient(socket.username, item);
```

Comment:
This is the only ref to `self` above. Should be able to just use `this` by using arrow function above vs `function(socket)`